src/components/LegacyMediaAnalytics/index.js

```diff
@@ -0,0 +1,119 @@
+import createMediaSessionCacheManager from "../MediaCollection/createMediaSessionCacheManager";
+import createTrackMediaEvent from "../MediaCollection/createTrackMediaEvent";
+import createTrackMediaSession from "../MediaCollection/createTrackMediaSession";
+import createMediaHelper from "./createMediaHelper";
+import {
+  AD_METADATA_KEYS as AdMetadataKeys,
+  AUDIO_METADATA_KEYS as AudioMetadataKeys,
+  EVENT as Event,
+  MEDIA_OBJECT_KEYS as MediaObjectKey,
+  MEDIA_TYPE as MediaType,
+  PLAYER_STATE as PlayerState,
+  STREAM_TYPE as StreamType,
+  VIDEO_METADATA_KEYS as VideoMetadataKeys
+} from "./constants/constants";
+import createGetInstance from "./createGetInstance";
+import { noop, uuid } from "../../utils";
+import createOnBeforeMediaEvent from "../MediaCollection/createOnBeforeMediaEvent";
+
+const createLegacyMediaAnalytics = ({
+  eventManager,
+  sendEdgeNetworkRequest,
+  config,
+  logger,
+  consent
+}) => {
+  const mediaSessionCacheManager = createMediaSessionCacheManager({ config });
+
+  const mediaEventManager = createMediaEventManager({
+    sendEdgeNetworkRequest,
+    config,
+    logger,
+    consent,
+    eventManager
+  });
+
+  const trackMediaEvent = createTrackMediaEvent({
+    mediaSessionCacheManager,
+    mediaEventManager,
+    config
+  });
+  const trackMediaSession = createTrackMediaSession({
+    config,
+    logger,
+    mediaEventManager,
+    mediaSessionCacheManager,
+    legacy: true
+  });
+  const onBeforeMediaEvent = createOnBeforeMediaEvent({
+    mediaSessionCacheManager,
+    logger,
+    config,
+    trackMediaEvent
+  });
+  return {
+    lifecycle: {
+      onBeforeEvent({ mediaOptions, onResponse = noop }) {
+        const { legacy, playerId, getPlayerDetails } = mediaOptions;
+
+        if (!legacy) {
+          return;
+        }
+        onResponse(({ response }) => {
+          return onBeforeMediaEvent({ playerId, getPlayerDetails, response });
+        });
+      }
+    },
+    commands: {
+      getMediaAnalyticsTracker: {
+        run: () => {
+          if (!config.mediaCollection) {
+            logger.warn("Media Collection is not configured.");
+          }
+
+          logger.info("Media Analytics Legacy Component was configured");
+
+          const mediaAnalyticsHelper = createMediaHelper({ logger });
+
+          return Promise.resolve({
+            getInstance: () => {
+              return createGetInstance({
+                logger,
+                trackMediaEvent,
+                trackMediaSession,
+                uuid
+              });
+            },
+            Event,
+            MediaType,
+            PlayerState,
+            StreamType,
+            MediaObjectKey,
+            VideoMetadataKeys,
+            AudioMetadataKeys,
+            AdMetadataKeys,
+            ...mediaAnalyticsHelper
+          });
+        }
+      }
+    }
+  };
+};
+
+createLegacyMediaAnalytics.namespace = "Legacy Media Analytics";
+
+export default createLegacyMediaAnalytics;
```

Comment:
There is some logic here. It would be great if you could split out lines 68-113 into a "createComponent" similar to the Identity component. That way you can unit test the logic in the component.